APMrover2/AP_Arming.cpp

```diff
@@ -185,3 +186,13 @@ bool AP_Arming_Rover::parameter_checks(bool report)
     return true;
 }
 
+// check if arming allowed from this mode
+bool AP_Arming_Rover::mode_checks(bool report)
+{
+    //display failure if arming is not allowed
+    if (!rover.control_mode->allows_arming_from_transmitter()) {
```

Comment:
OK, so this line doesn't look quite right.  You may not be arming from the transmitter, so this call is suspect.